controllers/volumereplication_controller.go

```diff
@@ -162,28 +171,46 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case replicationv1alpha1.Primary:
 		err := r.markVolumeAsPrimary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Secondary:
 		err := r.markVolumeAsSecondary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Resync:
 		err := r.resyncVolume(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	default:
+		r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, ("unsupported image state " + string(instance.Spec.ImageState)))
 		return ctrl.Result{}, fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
```

Comment:
Define a new error and reuse it at both places instead of duplicating.

Reply:
Done